src/main/java/se/sundsvall/seabloader/service/mapper/InvoiceMapper.java

```diff
@@ -76,8 +79,8 @@ public static InExchangeInvoiceStatusType toInExchangeInvoice(final String xml)
 	private static InvoicePdf toInvoicePdf(final InExchangeInvoiceStatusType inExchangeInvoiceStatusType, final OutputStream outputStream) {
 
 		if (Objects.isNull(inExchangeInvoiceStatusType.getOriginalInvoice()) || Objects.isNull(outputStream)) {
-			LOGGER.warn("OriginalInvoice or attachments not found in invoice with invoiceId: {}", inExchangeInvoiceStatusType.getInvoice().getInvoiceId());
-			return null;
+			LOGGER.error("OriginalInvoice or attachments not found in invoice with invoiceId: {}", inExchangeInvoiceStatusType.getInvoice().getInvoiceId());
+			throw Problem.valueOf(BAD_REQUEST, format("OriginalInvoice or attachments not found in invoice with invoiceId: %s", inExchangeInvoiceStatusType.getInvoice().getInvoiceId()));
```

Comment:
Why are you using a Problem with a BAD_REQUEST-status (client error) here? No status is sent back to any client right?

Why not just an IllegalArgumentException?